client/my-sites/stats/features/modules/stats-locations/stats-locations.tsx

```diff
@@ -77,15 +80,24 @@ const StatsLocations: React.FC< StatsModuleLocationsProps > = ( {
 	const supportUrl = isOdysseyStats
 		? `${ JETPACK_SUPPORT_URL_TRAFFIC }#views-by-locations`
 		: LOCATIONS_SUPPORT_URL;
-
-	const urlGeoMode =
-		initialGeoMode || new URLSearchParams( window.location.search ).get( 'geoMode' );
-	const [ selectedOption, setSelectedOption ] = useState(
-		urlGeoMode && urlGeoMode in GEO_MODES ? urlGeoMode : OPTION_KEYS.COUNTRIES
-	);
+	const [ selectedOption, setSelectedOption ] = useState( () => {
+		const urlGeoMode = query.geoMode ?? initialGeoMode;
+		return urlGeoMode && urlGeoMode in GEO_MODES ? urlGeoMode : OPTION_KEYS.COUNTRIES;
+	} );
 
 	const [ countryFilter, setCountryFilter ] = useState< string | null >( null );
 
+	useEffect( () => {
+		if ( ! summary ) {
+			return;
+		}
+
+		// If URL has valid param and it's different from state, update state
+		if ( query.geoMode !== selectedOption ) {
+			page( getPathWithUpdatedQueryString( { geoMode: selectedOption } ) );
```

Comment:
👍

Reply:
I would like to share [some thoughts on direct page navigation](https://github.com/Automattic/wp-calypso/compare/update/use-geo-mode-in-url...update/use_geo_mode_in_url_with_direct_navigation?expand=1) when changing the selected `geoMode` option, rather than redirecting after the page re-renders by local state change. 🙂